packages/rpc-methods/src/restricted/manageAccounts.test.ts

```diff
@@ -0,0 +1,198 @@
+import { SubjectType, PermissionType } from '@metamask/permission-controller';
+import { MOCK_SNAP_ID } from '@metamask/snaps-utils/test-utils';
+
+import {
+  methodName,
+  manageAccountsBuilder,
+  manageAccountsImplementation,
+  specificationBuilder,
+  validateParams,
+} from './manageAccounts';
+
+// To Do:
+// Move the class SnapKeyring to it's own module
+// add mock the method in this test instead of the entire class
+class SnapKeyringMock {
+  static type = 'Snap Keyring';
+
+  accounts: string[] = [];
+
+  handleKeyringSnapMessage = async (
+    _origin: string,
+    _params: any,
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    _saveSnapKeyring: Function,
+  ): Promise<any> => {
+    return true;
+  };
+}
+
+describe('specification', () => {
+  it('builds specification', () => {
+    const methodHooks = {
+      getSnapKeyring: jest.fn(),
+      saveSnapKeyring: jest.fn(),
+    };
+
+    expect(
+      specificationBuilder({
+        allowedCaveats: null,
+        methodHooks,
+      }),
+    ).toStrictEqual({
+      allowedCaveats: null,
+      methodImplementation: expect.anything(),
+      permissionType: PermissionType.RestrictedMethod,
+      targetName: methodName,
+      subjectTypes: [SubjectType.Snap],
+    });
+  });
+});
+
+describe('builder', () => {
+  it('has the expected shape', () => {
+    expect(manageAccountsBuilder).toMatchObject({
+      targetName: methodName,
+      specificationBuilder: expect.any(Function),
+      methodHooks: {
+        getSnapKeyring: true,
+        saveSnapKeyring: true,
+      },
+    });
+  });
+
+  it('builder outputs expected specification', () => {
+    expect(
+      manageAccountsBuilder.specificationBuilder({
+        methodHooks: {
+          getSnapKeyring: jest.fn(),
+          saveSnapKeyring: jest.fn(),
+        },
+      }),
+    ).toMatchObject({
+      permissionType: PermissionType.RestrictedMethod,
+      targetName: methodName,
+      allowedCaveats: null,
+      methodImplementation: expect.any(Function),
+    });
+  });
+});
+
+describe('manageAccountsImplementation', () => {
+  const mockCAIP10Account =
+    'eip155:1:0xab16a96D359eC26a11e2C2b3d8f8B8942d5Bfcdb';
+  const mockSnapId = MOCK_SNAP_ID;
```

Comment:
This seems redundant.
```suggestion
```

Reply:
Updated in [f126083](https://github.com/MetaMask/snaps/pull/1290/commits/f12608325bdf6a0ecfecf4a3bff49958cd3504de)